### PR TITLE
added dev tooltips when hovering over interactive buttons

### PR DESCRIPTION
**Pull request by LemonFace0309**
LOOM: https://www.loom.com/share/bf5ca5ed8ac345f79b55017822d62edf?sid=5665ccea-2aa6-4f83-996d-170c6a72b834

### CODE REVIEW

**Comment by LemonFace0309** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sutrolabs/census-embedded-demo/30?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#30** <a href="https://app.graphite.dev/github/pr/sutrolabs/census-embedded-demo/30?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#28** <a href="https://app.graphite.dev/github/pr/sutrolabs/census-embedded-demo/28?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @LemonFace0309 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: components/Source.js [hunk 1]
```diff
@@ -92,103 +96,144 @@ export default function Source({
...
-        className="relative z-50"
-      >
-        <div className="fixed inset-0 bg-black/30" aria-hidden="true" />
-        <div className="fixed inset-0 flex w-screen items-center justify-center p-4">
-          <Dialog.Panel className="flex w-full max-w-md flex-col gap-4 rounded bg-stone-100 px-8 py-6">
-            <Dialog.Title className="text-xl font-bold text-stone-700">Delete source</Dialog.Title>
-            <Dialog.Description className="text-stone-600">
-              This will permanently remove your connection to {label}.
-            </Dialog.Description>
+    <>
```
Comment (LemonFace0309) [inline components/Source.js:105]: no functional changes, just added tooltips at the end again

File: components/SyncObject.js [hunk 2]
```diff
@@ -2,135 +2,217 @@ import { useState } from "react"
...
-                }
-                const data = await response.json()
-                setSyncs((syncs) => syncs.map((item) => (item.id === sync.id ? data : item)))
-                refetchSyncs()
-              } finally {
-                setLoading(false)
-                setDisabledOverride()
-              }
-            }}
+    <>
```
Comment (LemonFace0309) [inline components/SyncObject.js:25]: no functional changes, just added tooltips at the end again

File: utils/url.js [hunk 3]
```diff
@@ -1,3 +1,3 @@
-export const censusBaseUrl = process.env["CENSUS_BASE_URL"] ?? "https://app.getcensus.com"
+export const censusBaseUrl = process.env["NEXT_PUBLIC_CENSUS_BASE_URL"] ?? "https://app.getcensus.com"
```
Comment (LemonFace0309) [inline utils/url.js:1]: this exposes the env variable to the frontend. Shouldn't matter here

File: package.json [hunk 4]
```diff
@@ -17,6 +17,8 @@
     "react": "^18.0.0",
     "react-chartjs-2": "^5.2.0",
     "react-dom": "^18.0.0",
+    "react-error-boundary": "^4.0.13",
```
Comment (yeldah3) [inline package.json:20]: Where is react-error-boundary being used?

Comment (LemonFace0309) [reply package.json:20]: oh shoot, this was the package I meant to delete, not react-tooltip LOL my bad!!

File: pages/integrations/import-dataset.js [hunk 5]
```diff
@@ -22,27 +22,36 @@ export default function ImportDataset({
   runs,
 }) {
   const [embedSourceFlow, setEmbedSourceFlow] = useState(true)
+  const [devMode, setDevMode] = useState(true)
```
Comment (yeldah3) [inline pages/integrations/import-dataset.js:25]: Mind double checking with Arjit that `true` is the right default here? I like it as true for discoverability, but I wonder if it causes too much cognitive overload with all the hovers

**Review by yeldah3 — COMMENTED** [review]
Code looks great!! Just want to confirm if `true` is the right default for dev mode. I lean no

**Review by yeldah3 — APPROVED** [review]

**Pull request merged**